### ISSUE TITLE
6.13.16 scour attempts to `lstat` a file deleted via another thread, race condition

### ISSUE THREAD

**Issue by akrherz** [Unidata/LDM#103]
I am using LDM 6.13.16 on Centos 8 Stream 64 bit.  I've noticed that since the upgrade to this release, I sometimes get errors like the following from `ldmadmin scour`

> 20211028T090217.262441Z scour[118662]               scour.c:scourFilesAndDirs:291       ERROR lstat("/data/gempak/nexrad/NIDS/LVX/N3K/N3K_20211026_0045") failed: No such file or directory

Out of deleting thousands of files, I only see one or two errors reported on some days, but not all. I know that you recently updated scour to use c code and not perl, perhaps there is some threading / race condition with how files are deleted?

The /data path is NFS mounted, so perhaps there is troubles there.  I verified that I am only running 1 scour process from cron and this is my scour.conf

```
/mesonet/data/gempak/model      1
/data/gempak/model				10
/data/gempak/nexrad				2
/data/gempak 					8
/data/rcm						7
/data/text						14
```

Thanks.

**Comment by sebenste** [comment]
I have seen this occur in the "old" way of scouring as well in LDM 6.13.10 and earlier, but it's now a moot point.

**Comment by akrherz** [comment]
Perhaps a command line switch could be offered to disable threaded scouring?  Or maybe this particular error could be sent to a lower priority log level?

**Comment by mustbei** [comment]
The new scour program spawns as many threads as there are directory entries (in scour.conf.) Therefore, to make it mono-threaded (without code change) it suffices to provide one directory entry at a time (to ensure non-concurrency.) It is also possible to enforce sequentiality with minor code change and a switch if warranted. Setting this error to a lower priority log level is also possible and only requires minimum code change.

**Comment by semmerson** [comment]
@akrherz Or one could modify their scour(1) configuration-file to avoid overlapping entries.

**Comment by akrherz** [comment]
> @akrherz Or one could modify their scour(1) configuration-file to avoid overlapping entries.

Agreed, but that is brittle as I may add a new folder and forget to add a custom entry for it and very annoying as I have to add one entry for each sub-folder.  Additionally, overlapping entries make total sense in my mind.

I have a blanket policy for anything in `/data/gempak` being at most 10 days old and then anything in `/data/gempak/nexrad` being at most 2 days old.
